src/modules/emote_menu/stores/index.js

```diff
@@ -0,0 +1,193 @@
+import Fuse from 'fuse.js';
+import SafeEventEmitter from '../../../utils/safe-event-emitter.js';
+import watcher from '../../../watcher.js';
+import emojiCategories from './emoji-categories.js';
+import emotes from '../../emotes/index.js';
+import Icons from '../components/Icons.jsx';
+import emoteStorage from './emote-storage.js';
+import {loadTwitchEmotes} from './twitch-emotes.js';
+import cdn from '../../../utils/cdn.js';
+import settings from '../../../settings.js';
+import {SettingIds} from '../../../constants.js';
+
+const MAXIMUM_VISIBLE_FREQUENTLY_USED = 27;
+
+const validateTotalColumns = () => (window.innerWidth <= 400 ? 7 : 9);
+
+function chunkArray(array, size) {
+  if (array.length <= size) {
+    return [array];
+  }
+  return [array.slice(0, size), ...chunkArray(array.slice(size), size)];
+}
+
+const fuse = new Fuse([], {
+  keys: ['code'],
+  shouldSort: true,
+  threshold: 0.3,
+});
+
+class EmoteStore extends SafeEventEmitter {
+  constructor() {
+    super();
+
+    this.rows = [];
+    this.headers = [];
+
+    this.totalCols = validateTotalColumns();
+
+    this.providers = {
+      dependable: [], // emote dependent emotes
+      extension: [], // channel dependent emotes
+      fixed: [], // emotes that wont change
+    };
+
+    watcher.on('channel.updated', async () => {
+      await this.loadFixedEmotes();
+      this.loadExtensionEmotes();
+      this.loadDependableEmotes();
+      this.createRows();
+    });
+
+    window.addEventListener('resize', () => {
+      const newTotalRows = validateTotalColumns();
+      if (newTotalRows !== this.totalCols) {
+        this.totalCols = newTotalRows;
+        this.createRows();
+      }
+    });
+
+    settings.on(`changed.${SettingIds.EMOTES}`, () => {
+      this.loadExtensionEmotes();
+      this.loadDependableEmotes();
+      this.createRows();
+    });
+  }
+
+  async loadFixedEmotes() {
+    if (this.providers.fixed.length > 0) return; // constants already been loaded
```

Comment:
if a user buys a subscription, their new emotes will never populate outside of a full page refresh.